packages/auth/src/accessToken/routes.ts

```diff
@@ -86,3 +88,22 @@ async function revokeToken(
     ctx.status = 204
   }
 }
+
+async function rotateToken(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  //TODO: verify accessToken with httpsig method
+  const result = await deps.accessTokenService.rotate(ctx.params['id'])
```

Comment:
I used [managementId](https://github.com/interledger/rafiki/blob/422f084017e3f24ddbde7c819b95a2f287fcd768/packages/auth/src/accessToken/model.ts#L23) so that management URLs wouldn't expose internal database ids. You can update the [migration file](https://github.com/interledger/rafiki/blob/dc-token-rotation/packages/auth/migrations/20220504163005_create_access_token_table.js) and add the `.unique()` modifier to it so you can call `.findOne()` with `managementId`.

Reply:
I updated it to use `managementId`, but should `managementId` be unique if we can have expired tokens for the same access token?

Reply:
From the GNAP spec [section 3.2.1](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-3.2.1) (emphasis mine):

> manage (string):  The management URI for this access token.  If
      provided, the client instance MAY manage its access token as
      described in [Section 6](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-6).  This management URI is a function of the
      AS and is separate from the RS the client instance is requesting
      access to.  **This URI MUST NOT include the access token value and
      SHOULD be different for each access token issued in a request.**
      OPTIONAL.

So we should be generating a new `managementId` each time a token is rotated. It might make sense to just delete the old token and return a newly created one (with corresponding `managementId`) but it could work either way.